### ISSUE TITLE
pre-alpha Tracing UX tuning

### ISSUE THREAD

**Issue by seanbrookes** [strongloop/strong-arc#1095]
the 1:1 UI/UX concurix to Arc module port is essentially complete but there are some UX interactions that could use some attention to call the implementation 'done' for alpha.
These include things like:
- [x]  improving the initial trace selection from the main timeline chart and transaction history charts on the front page
- [ ] add visual indicators to the transaction and waterfall histories to indicate relative time of exection for more relevant visual scanning
- [ ] add placeholder icons for landing page badge and inter-waterfall navigation on the waterfall detail view

pre-alpha occurred in https://github.com/strongloop/strong-arc/pull/1168

**Comment by seanbrookes** [comment]
this is a mopup(20%) story for #354 (80%)
- not all of the above will necessarily get complete by end of sprint but there is some low hanging fruit
- it is expected that after a more formal UX review changes will be requested so this should be considered realitvely cosmetic work

**Comment by chandadharap** [comment]
Wouldn't that be for beta? Alpha is your first 80% , we'd not release this
at alpha.

But we can add new stories to progress towards a beta. A beta can be
released.

Would you agree?
- chanda

On Friday, March 20, 2015, Sean Brookes <notifications@github.com> wrote:

> the 1:1 UI/UX concurix to Arc module port is essentially complete but
> there are some UX interactions that could use some attention to call the
> implementation 'done' for alpha.
>
> These include things like:
>
>    -  improving the initial trace selection from the main timeline chart
>    and transaction history charts on the front page
>    -  add visual indicators to the transaction and waterfall histories to
>    indicate relative time of exection for more relevant visual scanning
>    -  add placeholder icons for landing page badge and inter-waterfall
>    navigation on the waterfall detail view
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/strongloop/strong-arc/issues/1095>.
>


-- 
Chanda Dharap
Director, Engineering
@StrongLoop Inc
(510) 304-3924
chanda@strongloop.com

*StrongLoop <http://strongloop.com/>** makes it easy to **develop APIs
<http://strongloop.com/mobile-application-development/loopback/>** in Node,
plus get DevOps capabilities
<http://strongloop.com/node-js-performance/strongops/> like monitoring,
debugging and clustering**.*

**Comment by seanbrookes** [comment]
see scrum node-ops #354 for reference:

**Comment by seanbrookes** [comment]
We can call it anything but I won't be comfortable calling it 'alpha' until it is working end-to=end:
- functional UI (see #354)
- the renamed repo's are integrated
- the new pm api's are integtrated 
- the reworked host/app selector flow is integrated (see previous dependency)

This story is icing on top of #354 to close out this sprint

**Comment by chandadharap** [comment]
okay, sounds good.-c

On Fri, Mar 20, 2015 at 12:56 PM, Sean Brookes <notifications@github.com>
wrote:

> We can call it anything but I won't be comfortable calling it 'alpha'
> until it is working end-to=end:
>
>    - functional UI (see #354
>    <https://github.com/strongloop/strong-arc/issues/354>)
>    - the renamed repo's are integrated
>    - the new pm api's are integtrated
>    - the reworked host/app selector flow is integrated (see previous
>    dependency)
>
> This story is icing on top of #354
> <https://github.com/strongloop/strong-arc/issues/354> to close out this
> sprint
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/strongloop/strong-arc/issues/1095#issuecomment-84126810>
> .
>



-- 
Chanda Dharap
Director, Engineering
@StrongLoop Inc
(510) 304-3924
chanda@strongloop.com

*StrongLoop <http://strongloop.com/>** makes it easy to **develop APIs
<http://strongloop.com/mobile-application-development/loopback/>** in Node,
plus get DevOps capabilities
<http://strongloop.com/node-js-performance/strongops/> like monitoring,
debugging and clustering**.*

**Comment by seanbrookes** [comment]
I'm putting together some high level wireframe proposals to feed to the UX designer to wrap up the UX/design for target release
- wireframes are based on initial port of concurix UI to Arc

**Comment by seanbrookes** [comment]
UX is functional now with the following gaps:
- the interactions are still clunky when clicking between views so will be adding some light animations to smooth things out
- some header/breadcrumb display should be added to give the user better context around what they are seeing
- hover/highlight effect on waterfall detail view seems to have been lost and needs to be fixed 
- implement initial visual design

**Comment by chandadharap** [comment]
Can we make Initial VD a separate story?

On Thursday, April 2, 2015, Sean Brookes <notifications@github.com> wrote:

> UX is functional now with the following gaps:
>
>    - the interactions are still clunky when clicking between views so
>    will be adding some light animations to smooth things out
>    - some header/breadcrumb display should be added to give the user
>    better context around what they are seeing
>    - hover/highlight effect on waterfall detail view seems to have been
>    lost and needs to be fixed
>    - implement initial visual design
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/strongloop/strong-arc/issues/1095#issuecomment-88961492>
> .
>


-- 
Chanda Dharap
Director, Engineering
@StrongLoop Inc
(510) 304-3924
chanda@strongloop.com

*StrongLoop <http://strongloop.com/>** makes it easy to **develop APIs
<http://strongloop.com/mobile-application-development/loopback/>** in Node,
plus get DevOps capabilities
<http://strongloop.com/node-js-performance/strongops/> like monitoring,
debugging and clustering**.*

**Comment by seanbrookes** [comment]
yes, that's what I was thinking as well

**Comment by seanbrookes** [comment]
the three checkboxes in the description should all be separate stories as well

**Comment by seanbrookes** [comment]
the second 2 at least 
- add visual indicators to the transaction and waterfall histories to indicate relative time of exection for more relevant visual scanning
- add placeholder icons for landing page badge and inter-waterfall navigation on the waterfall detail view

**Comment by chandadharap** [comment]
Lets wait on creating additional tuning stories until we have a review of a landed feature on master.

**Comment by seanbrookes** [comment]
PR #1168

**Comment by sam-github** [comment]
@chandadharap do you want to verify? This isn't a complete feature, just a bit of the traces epic, I think it should be closed.

**Comment by chandadharap** [comment]
Went through this PR. Question: Does strongloop-internal/scrum-nodeops#351 cover everything in this PR that might be open?

In particular some issues I need info on:
1) This commit message says "nearly complete" https://github.com/strongloop/strong-arc/commit/fc890a0129b94afd4c7e2c2530435ad6a6991453
2) What is a demo for "training"? Did we release a branch for sales?  or is it referencing the hosted Concurix service still? Or is it re-codeding to point to the demo app but no hosted SaaS model.

https://github.com/strongloop/strong-arc/commit/9b9b64e0da72203f15b9e299518b1297e24df10d

3) Mostly functional on repos? Im assuming there are no open bugs and all dependencies are re-coded?
https://github.com/strongloop/strong-arc/commit/c2ed40bc8d8bae86fee18d96f0931e6e1889196e

If the above can be answered, i can close this.

**Comment by seanbrookes** [comment]
I just pushed what should be the last commit for this story.   

The commit notes you are reading are historical and have been addressed 

- https://github.com/strongloop-internal/scrum-nodeops/issues/351 pretty much sums up what's left on the front/middle
- there was another tracing epic assigned to Testsuo I think.  we talked about merging what's left of that one into this epic during scrum this morning.

just a reminder: QE, dev blogs, documentation, etc stories are all out of scope of the current epic (351)

**Comment by chandadharap** [comment]
@sean thanks. 

And yes typically how i run a feature for delivery is when you complete the feature, it goes through a QA cycle. At thatpoint in parallel I'll work with Issac for a release plan and make stories for doc and blogs, sometimes examples.

Those things are never part of the same story. But part of the Epic. (351 is an Epic in name only) the actual Epic is a Milestone and will get allotted more stories as we progress.

**Comment by chandadharap** [comment]
re-opening, and trying out before closing.

**Comment by chandadharap** [comment]
- Installed strongloop from staging.
- Got cloned the feature branch for pre-alpha-tracing

ran "slc arc" Could not figure  how to get to Traces. Whats the url?

**Comment by chandadharap** [comment]
The branch works. I was missing an updated strong-mesh-client and strong-mesh-models from staging.
